### PR TITLE
Adds additional dockerfile policies

### PR DESCRIPTION
**Pull request by rishidhamija**

### CODE REVIEW

File: pkg/policies/opa/rego/docker/docker_run/AC_DOCKER_0031.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+    "name": "pipInstallKeepingCachedPackages",
+    "file": "pipInstallKeepingCachedPackages.rego",
+    "policy_type": "docker",
+    "resource_type": "docker_run",
+    "template_args": {
+        "prefix": "",
+        "suffix": ""
```
Comment (harkirat22) [inline pkg/policies/opa/rego/docker/docker_run/AC_DOCKER_0031.json:8]: "name" is missing

File: pkg/policies/opa/rego/docker/docker_run/AC_DOCKER_0033json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    "resource_type": "docker_run",
+    "template_args": {
+        "prefix": "",
+        "suffix": ""
+    },
+    "severity": "MEDIUM",
+    "description": "Ensure package version is specified to avoid failures",
+    "reference_id": "AC_DOCKER_0031",
+    "category": "Infrastructure Security",
+    "id": "AC_DOCKER_0031",
```
Comment (Rchanger) [inline pkg/policies/opa/rego/docker/docker_run/AC_DOCKER_0033json:14]: Duplicate id and reference id,  `pipInstallKeepingCachedPackages` policy also have the same id. Typo in filename `pkg/policies/opa/rego/docker/docker_run/AC_DOCKER_0033json`

File: pkg/policies/opa/rego/docker/docker_run/AC_DOCKER_0036.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    "file": "multipleRunAddCopyInstructions.rego",
+    "policy_type": "docker",
+    "resource_type": "docker_run",
+    "template_args": {
+        "prefix": "",
+        "suffix": "",
+        "name": "multipleRunAddCopyInstructions"
+    },
+    "severity": "LOW",
+    "description": "Ensure multiple commands (RUN, Copy, And) should be grouped in order to reduce the number of layers.",
```
Comment (Rchanger) [inline pkg/policies/opa/rego/docker/docker_run/AC_DOCKER_0036.json:12]: a typo here it should be Add not And

File: pkg/policies/opa/rego/docker/docker_run/multipleRunAddCopyInstructions.rego [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    command := item[_]
+    
+    instructions := {"copy", "add", "run"}
+	some i
+	check := [x | item[i].config == instructions[y]; x := item[i]]
+    
+    some j, k
+	Counter := [x | check[j].line - check[k].line == -1; x := check[j]]
+    
+    upperName := upper(instructions[y])
```
Comment (gaurav-gogia) [inline pkg/policies/opa/rego/docker/docker_run/multipleRunAddCopyInstructions.rego:24]: indentation issue

File: pkg/policies/opa/rego/docker/docker_run/dnfCleanAllMissing.rego [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+		"dnf in",
+		"dnf reinstall",
+		"dnf rei",
+		"dnf install-n",
+		"dnf install-na",
+		"dnf install-nevra",
+	]
+
+	some cmd
+	install := indexof(config, installCommands[cmd])
```
Comment (gaurav-gogia) [inline pkg/policies/opa/rego/docker/docker_run/dnfCleanAllMissing.rego:41]: use 4 spaces for indentation

**Review by gaurav-gogia — DISMISSED** [review]

**Review by harkirat22 — APPROVED** [review]

**Pull request merged**